tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -978,16 +1036,38 @@ private HttpURLConnection deleteBucketRestCall(String bucketUri) throws Exceptio
         TestCaseOptions.defaults()).execute();
   }
 
-  private void createObjectRestCall(String objectUri, byte[] objectContent, String md5,
-      Map<String, String> params) throws Exception {
+  private String computeObjectChecksum(byte[] objectContent) throws Exception {
+    MessageDigest md5Hash = MessageDigest.getInstance("MD5");
+    byte[] md5Digest = md5Hash.digest(objectContent);
+    return BaseEncoding.base64().encode(md5Digest);
+  }
+
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params) throws Exception {
+    createObjectRestCall(objectUri, objectContent, params, computeObjectChecksum(objectContent));
+  }
+
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params, String md5) throws Exception {
+    createObjectRestCall(objectUri, objectContent, params, md5, null);
+  }
+
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params,
+                                    Map<String, String> extraHeaders) throws Exception {
+    createObjectRestCall(objectUri, objectContent, params,
+        computeObjectChecksum(objectContent), extraHeaders);
+  }
+
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params,
+                                    String md5, Map<String, String> extraHeaders) throws Exception {
```

Comment:
Java-native "optional" parameters as opposed to callers specifying `null`. If this is too verbose then I can use a single method with two null parameter checks like previously.

Reply:
I decided to axe the method overloading for the verbosity mentioned above.